File: LeetCode Solutions/Problems/2347. Best Poker Hand.java
class Solution {
    public String bestHand(int[] ranks, char[] suits) {
        int flag = -1;
        int arr[] = new int[5];
        int arr1[] = new int[14];
        for (int i=0; i<5; i++){
            arr[suits[i]-'a']++;
            arr1[ranks[i]]++;
        }
        for (int i=0; i<5; i++){
            if (arr[i] >= 5){
                return "Flush";
            }
        }
        for (int i=0; i<14; i++){
            if (arr1[i] >= 3){
                flag = 0;
                break;
            }
            if (arr1[i] == 2){
                flag = 1;
            }
        }
        if (flag == 0){
            return "Three of a Kind";
        }
        else if (flag == 1){
            return "Pair";
        }
        return "High Card";
    }
}

File: LeetCode Solutions/Problems/2347. Best Poker Hand.cpp
class Solution {
public:
    string bestHand(vector<int>& ranks, vector<char>& suits) {
        if (suits[0] == suits[1] && suits[1] == suits[2] && suits[2] == suits[3] && suits[3] == suits[4]) {
            return "Flush";
        }
        sort(ranks.begin(), ranks.end());
        bool flag = false;
        for (int i=0; i<4; ++i) {
            if (ranks[i] == ranks[i+1] && ranks[i] == ranks[i+2]) {
                return "Three of a Kind";
            }
            if (ranks[i] == ranks[i+1]) {
                flag = true;
            }    
        }
        if (flag) {
            return "Pair";
        }
        return "High Card";
    }
};